### DISCUSSION TITLE
Managing seasonal products.

### DISCUSSION THREAD

**Discussion by holden** [swellstores/community#321] [Help] [answered]
Hi,

We have some items which we only sell seasonally, eg Xmas, etc. We don't want to make them inactive, as we'd still like people to be able to see that they exist. They're digital products, we'd just like a way to prevent people from people from making purchases for 10-months out of the year.  

I tried setting the `stock_status` to out-of-stock but it says this is read-only.

What is the easiest way to disable the add-to-cart option?

Thanks,

**Comment by steven-k1** [comment]
@holden When attempting to set the `stock_status`, was this done via the frontend API? If so, this is most likely a permissions restriction.

By setting the item out of stock, though, this should prevent customers from adding the item to the cart as long as you do not have the 'Allow purchase when out of stock' setting enabled.

**Reply by holden** [reply]
No, this was using the backend API when trying to update an existing product. Yeah, I figured as much I just can't get it to work. The update works fine when I change any of the other attributes on the product, just not the `stock_status`.

`{"errors":{"stock_status":{"message":"Read-only field: stock_status","code":"READONLY"}}}`

I tried a variety of things but it doesn't like it:

```
swell_json.merge!({
  stock_purchasable: false,
  stock_backorder: false,
  stock_status: 'out_of_stock'
})
```

**Reply by ericingram** [reply, marked as answer]
@holden `stock_status` is an auto enum field that's not supposed to be set directly. You can see the fields specifically by making this request in the API console: 

```
GET /:models/products/fields/stock_status
```

Generally it is set based on stock records of the product, but if you find a case where we should support setting this value directly, we would consider it :)
